google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -494,6 +489,16 @@ public GapicSpannerRpc(final SpannerOptions options) {
     }
   }
 
+  private static HeaderProvider headerProviderWithUserAgentFrom(HeaderProvider headerProvider) {
+    final Map<String, String> headersWithUserAgent = new HashMap<>(headerProvider.getHeaders());
+    final String userAgent = headersWithUserAgent.get(USER_AGENT_KEY);
```

Comment:
@thiagotnunes `user-agent` and `User-Agent` should be equivalent.
This needs case insensitivity.

Reply:
Hey there, thanks for pointing this out.

This has been fixed in the following versions:

1. [3.3.3](https://github.com/googleapis/java-spanner/blob/3.3.x/CHANGELOG.md#333-2021-05-07)
1. [4.0.3](https://github.com/googleapis/java-spanner/blob/4.0.x/CHANGELOG.md#403-2021-05-03)
1. [5.2.1](https://github.com/googleapis/java-spanner/blob/5.2.x/CHANGELOG.md#521-2021-05-03)
1. [6.4.0](https://github.com/googleapis/java-spanner/blob/master/CHANGELOG.md#640-2021-04-29)